e2e/app/run.go

```diff
@@ -83,49 +85,34 @@ func Deploy(ctx context.Context, def Definition, cfg DeployConfig) (*pingpong.XD
 
 	contracts.UseStagingOmniRPC(def.Testnet.BroadcastOmniEVM().ExternalRPC)
 
-	if err := fundAnvilAccounts(ctx, def); err != nil {
-		return nil, err
-	}
-
-	if err := deployAllCreate3(ctx, def); err != nil {
-		return nil, err
+	// Prep for deploying contracts.
+	var eg1 errgroup.Group
+	eg1.Go(func() error { return fundAnvilAccounts(ctx, def) })
+	eg1.Go(func() error { return deployAllCreate3(ctx, def) })
+	if err := eg1.Wait(); err != nil {
+		return nil, errors.Wrap(err, "deploy prep")
 	}
 
+	// Deploy portals
 	if err := def.Netman().DeployPortals(ctx, genesisValSetID, genesisVals); err != nil {
-		return nil, err
-	}
-	logRPCs(ctx, def)
-
-	if err := initPortalRegistry(ctx, def); err != nil {
-		return nil, err
-	}
-
-	if err := allowStagingValidators(ctx, def); err != nil {
-		return nil, err
-	}
-
-	if def.Testnet.Network.IsEphemeral() {
-		if err := DeployGasApp(ctx, def); err != nil {
-			return nil, err
-		}
+		return nil, errors.Wrap(err, "deploy portals")
 	}
 
-	if err := DeployBridge(ctx, def); err != nil {
-		return nil, errors.Wrap(err, "setup token bridge")
-	}
-
-	if err := maybeSubmitNetworkUpgrade(ctx, def); err != nil {
-		return nil, err
-	}
-
-	if err := FundValidatorsForTesting(ctx, def); err != nil {
-		return nil, err
-	}
+	logRPCs(ctx, def)
 
+	// Deploy other contracts (and other on-chain setup)
+	var eg2 errgroup.Group
+	eg2.Go(func() error { return initPortalRegistry(ctx, def) })
+	eg2.Go(func() error { return allowStagingValidators(ctx, def) })
+	eg2.Go(func() error { return DeployEphemeralGasApp(ctx, def) })
+	eg2.Go(func() error { return DeployBridge(ctx, def) })
+	eg2.Go(func() error { return maybeSubmitNetworkUpgrade(ctx, def) })
+	eg2.Go(func() error { return FundValidatorsForTesting(ctx, def) })
 	if def.Manifest.DeploySolve {
-		if err := solve.DeployContracts(ctx, NetworkFromDef(def), def.Backends()); err != nil {
-			return nil, errors.Wrap(err, "deploy solve contracts")
-		}
+		eg2.Go(func() error { return solve.DeployContracts(ctx, NetworkFromDef(def), def.Backends()) })
+	}
+	if err := eg2.Wait(); err != nil {
```

Comment:
Just for my understanding: this refactoring assumes that there is no sequential dependency on these deployments, right? Clarifying because `initPortalRegistry` being the first deployment might indicate that there are some dependencies.

Reply:
indeed, no dependencies.